File: leet_1822_sign_of_product.c
int arraySign(int* nums, int numsSize)
{
    int i = -1, a = 1 ;

    while (++i < numsSize)
    {
        if (!nums[i])   return 0 ;
        if (nums[i]<0)  a=-a ;
        
    }
    return a < 0 ? -1 : 1 ;
}

File: leet_1822_sign_of_product.ts
function arraySign(nums: number[]): number {
    let sign = 1
    for (let n of nums) {
        if (n == 0) {
            return 0
        }
        if (n < 0) {
            sign = -sign
        }
    }
    return sign > 0 ? 1 : -1
};
